api/apps/geoprocessing/src/migrations/geoprocessing/1655901622494-AddApiFeatureIdToScenarioFeaturesData.ts

```diff
@@ -0,0 +1,36 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddApiFeatureIdToScenarioFeaturesData1653916456540
+  implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      DELETE FROM scenario_features_data;
+    `);
+
+    await queryRunner.query(`
+      ALTER TABLE scenario_features_data
+      ADD COLUMN api_feature_id uuid NOT NULL;
+    `);
+
+    await queryRunner.query(`
+      alter table scenario_features_preparation
+      ADD COLUMN api_feature_id uuid NOT NULL;
+    `);
+  }
```

Comment:
I think it was me who advised that nuking `scenario_features_data` would be ok, but now seeing this in code, I'm not so sure anymore - true, there's no _real_ production instance, but I think we should try to disrupt existing "public" instances as little as possible.

In a way, given that so far we only have `copy/v1` operations used in the wild (if any `split/v1` has ever been attempted, we can safely consider these as not properly done and discardable), this can work to our advantage: what about adding the `api_feature_id` as nullable first, then setting it to the actual expected value via something like `update scenario_features_data set api_feature_id = (select fd.feature_id from scenario_features_data sfd inner join features_data fd on fd.id = sfd.feature_class_id where sfd.id = scenario_features_data.id)`?

and finally setting the new column as non-nullable.

in theory we should not need to repeat this for `scenario_features_preparation` unless some stale data was left around in an instance's geodb through some preparation step that was interrupted (though IIRC prep is done in a db transaction so it may be guaranteed to complete or be rolled back without leaving stale data around).